### PR TITLE
fix(core-notification): throw console warn on dismissable error/warning

### PR DESCRIPTION
**Pull request by marcod1419**
This PR will cause a console warning to be thrown when `dismissible` is used in combination with the `error` or `warning` variants. Additionally, it improves the overall documentation of Notification.

### CODE REVIEW

File: packages/Notification/Notification.md [hunk 1]
```diff
@@ -68,7 +76,14 @@ Use the `warning` variant to provide feedback of a warning; users would still be
...
 ### Dismissible
 
-Use the `dismissible` prop to allow users to dismiss the `Notification` at any time. The `dismissible` prop must be used together with the `dismissibleA11yLabel` to remain accessible. Once dismissed, the `Notification` is removed from the DOM.
+Notifications can either be passive or require action from the user. If the information is passive then it should be dismissible as indicated by an interactive close button on the far right. If the Notification requires an action then make the option(s) clear with a Chevron Link or Paragraph link style.
+
+Use the dismissible prop to allow users to dismiss the Notification at any time. The `dismissible` prop must be used together with the `dismissibleA11yLabel` to remain accessible. Once dismissed, the Notification is removed from the DOM.
+
+Please note:
+
+- Branded, General and Success and Instructional variants can be dismissed
```
Comment (jraff) [inline packages/Notification/Notification.md]: Lets remove **"and Instructional"**, why are we capitalizing General, and Success?

File: packages/Notification/Notification.md [hunk 2]
```diff
@@ -68,7 +76,14 @@ Use the `warning` variant to provide feedback of a warning; users would still be
...
+- Error and Warning variants should not be dismissed
```
Comment (jraff) [inline packages/Notification/Notification.md]: Why are we capitalizing Warning?

File: packages/Notification/Notification.md [hunk 3]
```diff
@@ -1,8 +1,14 @@
 **Use the "Open isolated" button above to view this component in full-width mode.**
 
-The `Notification` spans the entire width of the screen, and aligns the message with the page content.
+A banner that highlights important messages.
 
-By default, notifications will be displayed in the `instructional` variant.
+Notifications are used to let users know:
+
+- System status such as an error or outage
+- Their actions has triggered a change such as confirmation of a payment
```
Comment (jraff) [inline packages/Notification/Notification.md]: Change "has" to "have"

Result: "Their actions have triggered a change such as confirmation of a payment"

File: packages/Notification/Notification.md [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+- Information such as letting the user know they have almost used all their data
+
+Colours help indicate the type of message. For errors use a red background. For other notifications use grey. Lastly, if the message is promotional use branded colours. These are reflected in code via the `variants` prop, and the background colour will change to match the desiered variant. For example, the `error` variant will produce a red background.
```
Comment (jraff) [inline packages/Notification/Notification.md]: Change `variants` to `variant`

File: packages/Notification/Notification.md [hunk 5]
```diff
@@ -17,6 +23,8 @@ By default, notifications will be displayed in the `instructional` variant.
   not use other complex components or paragraphs.
 - Use complete sentences, while keeping the content succinct.
 - Dismissible notifications should only be used for branded or general style notifications where there is no action needed from the user.
+- If the `variant` is `success` or `error`, there should be a `Box` with `between` level 3 between the icon and the text
```
Comment (jraff) [inline packages/Notification/Notification.md]: I don't think this line should belong on our docs

File: packages/Notification/Notification.md [hunk 6]
```diff
@@ -17,6 +23,8 @@ By default, notifications will be displayed in the `instructional` variant.
...
+- If the `variant` is `dismissible`, there should be a `Box` with `between` level 3 between the text and the dismiss button
```
Comment (jraff) [inline packages/Notification/Notification.md]: I don't think this line should belong on our docs

File: packages/Notification/Notification.md [hunk 7]
```diff
@@ -68,7 +76,14 @@ Use the `warning` variant to provide feedback of a warning; users would still be
 
 ### Dismissible
 
-Use the `dismissible` prop to allow users to dismiss the `Notification` at any time. The `dismissible` prop must be used together with the `dismissibleA11yLabel` to remain accessible. Once dismissed, the `Notification` is removed from the DOM.
+Notifications can either be passive or require action from the user. If the information is passive then it should be dismissible as indicated by an interactive close button on the far right. If the Notification requires an action then make the option(s) clear with a Chevron Link or Paragraph link style.
+
+Use the dismissible prop to allow users to dismiss the Notification at any time. The `dismissible` prop must be used together with the `dismissibleA11yLabel` to remain accessible. Once dismissed, the Notification is removed from the DOM.
```
Comment (jraff) [inline packages/Notification/Notification.md]: Can we add backticks around dismissible?

File: packages/Notification/Notification.md [hunk 8]
```diff
@@ -17,6 +23,8 @@ By default, notifications will be displayed in the `instructional` variant.
   not use other complex components or paragraphs.
 - Use complete sentences, while keeping the content succinct.
```
Comment (jraff) [inline packages/Notification/Notification.md]: "When there are multiple notifications, the messages should be served top to bottom from most to least important. Remove any unnecessary information such as promotional messages that add to the noise"

Should be added after this line

File: packages/Notification/Notification.md (hunk 1 above)
Comment (marcod1419) [reply packages/Notification/Notification.md]: Instructional is a variant type though. It's not currently deprecated, should it be?

File: packages/Notification/Notification.md (hunk 2 above)
Comment (marcod1419) [reply packages/Notification/Notification.md]: I decided to capitalize all variant types. I'm flat hand on keeping that though

File: packages/Notification/Notification.md (hunk 5 above)
Comment (marcod1419) [reply packages/Notification/Notification.md]: This was part of the misunderstanding of this ticket, I'll remove it.

File: packages/Notification/Notification.md (hunk 1 above)
Comment (jraff) [reply packages/Notification/Notification.md]: My mistake.

**Pull request merged**